test/org/sagebionetworks/bridge/validators/ConsentSignatureValidatorTest.java

```diff
@@ -0,0 +1,232 @@
+                "   \"imageData\":\"\",\n" +
+                "   \"imageMimeType\":\"image/fake\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        assertValidatorMessage(validator, sig, "imageData", "cannot be an empty string");
+    }
+
+    @Test
+    public void jsonEmptyImageMimeType() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageData\":\"" + TestConstants.DUMMY_IMAGE_DATA + "\",\n" +
+                "   \"imageMimeType\":\"\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        assertValidatorMessage(validator, sig, "imageMimeType", "cannot be an empty string");
+    }
+
+    @Test
+    public void jsonImageDataWithoutMimeType() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageData\":\"" + TestConstants.DUMMY_IMAGE_DATA + "\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void jsonImageMimeTypeWithoutData() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageMimeType\":\"image/fake\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void minAgeLimitButNoBirthdate() {
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").build();
+        assertValidatorMessage(validator, sig, "birthdate", "cannot be missing, null, or blank");
+    }
+    
+    @Test
+    public void minAgeLimitButBirthdateTooRecent() {
+        String birthdate = DateTime.now().minusYears(18).plusDays(1).toLocalDate().toString();
+        validator = new ConsentSignatureValidator(18);
```

Comment:
Actually, I think we want this the other way around. On Feb 28 2022, this *definitely* will fail, because Joda thinks there are exactly 18 years between Feb 29 2004 and Feb 28 2022. I think using fixed timestamps will make the test more reliable.

Reply:
I fixed the time to 2/28/2022, this does not seem to make any difference. If you do not remove a day, it succeeds, otherwise it fails as intended. It seems smart enough to deal with leap years.